src/OpenTelemetry.Shared/LocalFileBlob.cs

```diff
@@ -0,0 +1,121 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.IO;
+
+namespace OpenTelemetry.Shared
+{
+    /// <summary>
+    /// The <see cref="LocalFileBlob"/> allows to save a blob
+    /// in file storage.
+    /// </summary>
+    public class LocalFileBlob : IPersistentBlob
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LocalFileBlob"/>
+        /// class.
+        /// </summary>
+        /// <param name="fullPath">Absolute file path of the blob.</param>
+        public LocalFileBlob(string fullPath)
+        {
+            this.FullPath = fullPath;
+        }
+
+        public string FullPath { get; private set; }
+
+        /// <inheritdoc/>
+        public byte[] Read()
+        {
+            try
+            {
+                return File.ReadAllBytes(this.FullPath);
+            }
+            catch (Exception ex)
+            {
+                SharedEventSource.Log.Warning($"Reading a blob from file {this.FullPath} has failed.", ex);
+            }
+
+            return null;
+        }
+
+        /// <inheritdoc/>
+        public IPersistentBlob Write(byte[] buffer, int leasePeriodMilliseconds = 0)
+        {
+            string path = this.FullPath + ".tmp";
+
+            try
+            {
+                File.WriteAllBytes(path, buffer);
+
+                if (leasePeriodMilliseconds > 0)
+                {
+                    var timestamp = DateTime.Now.ToUniversalTime() + TimeSpan.FromMilliseconds(leasePeriodMilliseconds);
+                    this.FullPath += $"@{timestamp:yyy-MM-ddTHHmmss.ffffff}.lock";
+                }
+
+                File.Move(path, this.FullPath);
+            }
+            catch (Exception ex)
+            {
+                SharedEventSource.Log.Warning($"Writing a blob to file {path} has failed.", ex);
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public IPersistentBlob Lease(int leasePeriodMilliseconds)
+        {
+            var path = this.FullPath;
+            var leaseTimestamp = DateTime.Now.ToUniversalTime() + TimeSpan.FromMilliseconds(leasePeriodMilliseconds);
+            if (path.EndsWith(".lock", StringComparison.OrdinalIgnoreCase))
+            {
+                path = path.Substring(0, path.LastIndexOf('@'));
+            }
+
+            path += $"@{leaseTimestamp:yyy-MM-ddTHHmmss.ffffff}.lock";
+
+            try
+            {
+                File.Move(this.FullPath, path);
+            }
+            catch (Exception ex)
+            {
+                SharedEventSource.Log.Warning($"Acquiring a lease to file {this.FullPath} has failed.", ex);
+            }
+
+            this.FullPath = path;
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public void Delete()
+        {
+            try
+            {
+                if (File.Exists(this.FullPath))
```

Comment:
Is there a need to check `File.Exists`? (even if it returned true, other threads/proceeses might have deleted the file)

Reply:
Removed check for `File.Exists`. If a file is removed or locked by other thread we could just catch an exception to pass it on.